app/middleware/postcodeLookup.js

```diff
@@ -0,0 +1,52 @@
+const log = require('../lib/logger');
+const PostcodesIOClient = require('postcodesio-client');
+
+// rewire (a framework for mocking) doesn't support const
+// eslint-disable-next-line no-var
+var PostcodesIO = new PostcodesIOClient();
+// eslint-disable-next-line no-var
+var renderer = require('./renderer');
+
+function toArray(countries) {
+  return Array.isArray(countries) ? countries : [countries];
+}
+
+function isOutcode(postcodeDetails) {
+  return !postcodeDetails.incode;
+}
+
+function postcodeDetailsMapper(postcodeDetails) {
+  return {
+    isOutcode: isOutcode(postcodeDetails),
+    location: {
+      lat: postcodeDetails.latitude,
+      lon: postcodeDetails.longitude
+    },
+    countries: toArray(postcodeDetails.country)
+  };
+}
+
+async function lookupPostcode(req, res, next) {
+  const postcodeSearch = res.locals.location;
```

Comment:
Does the variable name need to be changed here? I'd prefer to see `const location = res.locals.location;` for ease of understanding.